fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/tree/lowcardinality/DecodeCollector.java

```diff
@@ -418,11 +418,10 @@ public DecodeInfo visitPhysicalJoin(OptExpression optExpression, DecodeInfo cont
         if (!result.inputStringColumns.containsAny(onColumns)) {
             return result;
         }
+        onColumns.getStream().forEach(c -> disableRewriteStringColumns.union(c));
         result.outputStringColumns.clear();
         result.inputStringColumns.getStream().forEach(c -> {
-            if (onColumns.contains(c)) {
-                disableRewriteStringColumns.union(c);
-            } else {
+            if (!onColumns.contains(c)) {
                 result.outputStringColumns.union(c);
```

Comment:
why need change this? if `InputStringColumns` doesn't contains onColumns, the onColumns must not be a low-cardinality string

Reply:
When the Join's child has inserted `DecodeNode` instead, `InputStringColumns ` is cleared, but we still need to disable the already inserted `Decode Column`, otherwise decode columns of join on-predicate may disturb join's required distributions(encode/decoded column's distributions are different).